### PR TITLE
feat(DMS): add dms rabbitmq flavors data source

### PR DESCRIPTION
**Pull request by houpeng80**
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:
   add dms rabbitmq flavors data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->

```release-note
   add dms rabbitmq flavors data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dms/' TESTARGS='-run TestAccRabbitMQFlavorsDataSource_basic'       
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccRabbitMQFlavorsDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccRabbitMQFlavorsDataSource_basic
=== PAUSE TestAccRabbitMQFlavorsDataSource_basic
=== CONT  TestAccRabbitMQFlavorsDataSource_basic
--- PASS: TestAccRabbitMQFlavorsDataSource_basic (8.53s)
PASS                                                                                                                                                                                                                               
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       8.590s
```

### CODE REVIEW

File: docs/data-sources/dms_rabbitmq_flavors.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+### Query the rabbitmq flavor details of the specified ID
+
+```hcl
+data "huaweicloud_dms_rabbitmq_flavors" "test" {
+  flavor_id = "c6.2u4g.cluster"
+}
+```
+
+### Query list of rabbitmq flavors that available in the availability zone list
```
Comment (deer-hang) [inline docs/data-sources/dms_rabbitmq_flavors.md]: make the title of `example usage` coincident.

Comment (houpeng80) [reply docs/data-sources/dms_rabbitmq_flavors.md]: done

File: docs/data-sources/dms_rabbitmq_flavors.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+subcategory: "Distributed Message Service (DMS)"
+---
+
+# huaweicloud_dms_rabbitmq_flavors
+
+Use this data source to get the list of rabbitMQ available flavor details within HuaweiCloud.
+
+## Example Usage
+
+### Query the list of rabbitmq flavors by cluster type
```
Comment (Lance52259) [inline docs/data-sources/dms_rabbitmq_flavors.md]: s/rabbitmq/rabbitMQ
MQ is the abbreviation of message queue

File: huaweicloud/services/acceptance/dms/data_source_huaweicloud_dms_rabbitmq_flavors_test.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  test_refer   = data.huaweicloud_dms_rabbitmq_flavors.baisc.flavors[0]
+  test_results = data.huaweicloud_dms_rabbitmq_flavors.test
+}
+
+output "type_validation" {
+  value = contains(local.test_results.flavors[*].type, local.test_refer.type)
+}
+
+output "arch_types_validation" {
+  value = !contains([for a in local.test_results.flavors[*].arch_types : contains(a, local.test_refer.arch_types[0])], false)
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dms/data_source_huaweicloud_dms_rabbitmq_flavors_test.go]: `alltrue()` will be better

File: docs/data-sources/dms_rabbitmq_flavors.md (hunk 2 above)
Comment (houpeng80) [reply docs/data-sources/dms_rabbitmq_flavors.md]: done

File: huaweicloud/services/acceptance/dms/data_source_huaweicloud_dms_rabbitmq_flavors_test.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/dms/data_source_huaweicloud_dms_rabbitmq_flavors_test.go]: done

**Pull request merged**